legal-api/src/legal_api/services/filings/validations/continuation_in.py

```diff
@@ -0,0 +1,173 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validation for the Continuation In filing."""
+from http import HTTPStatus  # pylint: disable=wrong-import-order
+from typing import Final, Optional
+
+from flask_babel import _ as babel  # noqa: N813, I004, I001, I003
+
+from legal_api.errors import Error
+from legal_api.models import Business, PartyRole
+from legal_api.services.filings.validations.common_validations import (
+    validate_court_order,
+    validate_foreign_jurisdiction,
+    validate_name_request,
+    validate_pdf,
+    validate_share_structure,
+)
+from legal_api.services.filings.validations.incorporation_application import (
+    validate_incorporation_effective_date,
+    validate_offices,
+    validate_parties_mailing_address,
+    validate_parties_names,
+)
+from legal_api.services.utils import get_date, get_str
+from legal_api.utils.legislation_datetime import LegislationDatetime
+
+
+def validate(filing_json: dict) -> Optional[Error]:  # pylint: disable=too-many-branches;
+    """Validate the Continuation In filing."""
+    filing_type = 'continuationIn'
+    if not filing_json:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid filing is required.')}])
+    msg = []
+
+    legal_type_path = '/filing/continuationIn/nameRequest/legalType'
+    legal_type = get_str(filing_json, legal_type_path)
+    if not legal_type:
+        msg.append({'error': babel('Legal type is required.'), 'path': legal_type_path})
+        return msg  # Cannot continue validation without legal_type
+
+    msg.extend(validate_name_request(filing_json, legal_type, filing_type))
+    msg.extend(validate_offices(filing_json, filing_type))
+    msg.extend(validate_roles(filing_json, legal_type, filing_type))
+
+    # FUTURE: this should be removed when COLIN sync back is no longer required. This names validation is required
+    # to work around first and middle name length mismatches between LEAR and COLIN. Syncing back to COLIN
+    # would error out on first and middle name length exceeding 20 characters for completing party
+    err = validate_parties_names(filing_json, filing_type)
+    if err:
+        msg.extend(err)
+
+    err = validate_parties_mailing_address(filing_json, legal_type, filing_type)
+    if err:
+        msg.extend(err)
+
+    err = validate_share_structure(filing_json, filing_type)
+    if err:
+        msg.extend(err)
+
+    msg.extend(validate_continuation_in_authorization(filing_json, filing_type))
+    msg.extend(_validate_foreign_jurisdiction(filing_json, filing_type, legal_type))
+
+    err = validate_incorporation_effective_date(filing_json)
+    if err:
+        msg.extend(err)
+
+    msg.extend(validate_continuation_in_court_order(filing_json, filing_type))
+
```

Comment:
Should also validate `filing.continuationIn.business` object?

If the existing business is an expro then this object will contain foundingDate, identifier and legalName.

You may not need to do anything with this object. It was populated by business search and/or colin fetch and the user is not able to edit any of the data.

Reply:
there is a separate ticket for that https://app.zenhub.com/workspaces/entities---olga-65af15f59e89f5043c2911f7/issues/gh/bcgov/entity/21104

Reply:
Perfect. Resolving.